mne_bids/write.py

```diff
@@ -1881,6 +1851,30 @@ def write_raw_bids(raw, bids_path, events_data=None, event_id=None,
                              f'Please use one of {CONVERT_FORMATS[datatype]} '
                              f'for {datatype} datatype.')
 
+    # raise error when trying to copy files (copyfile_*) into same location
+    # (src == dest, see https://github.com/mne-tools/mne-bids/issues/867)
+    if bids_path.fpath.exists() and not convert and \
+            bids_path.fpath.as_posix() == Path(raw_fname).as_posix():
+        raise RuntimeError(f'Desired output BIDSPath ("{bids_path.fpath}") is'
+                           ' the source file. Please pass a different output'
+                           ' BIDSPath, or set `format` to something other'
+                           ' than "auto".')
+
+    # otherwise if the BIDSPath currently exists, check if we
+    # would like to overwrite the existing dataset
+    if bids_path.fpath.exists():
+        if overwrite:
+            # Need to load data before removing its source
+            raw.load_data()
+            if bids_path.fpath.is_dir():
+                shutil.rmtree(bids_path.fpath)
+            else:
+                bids_path.fpath.unlink()
+        else:
+            raise FileExistsError(
```

Comment:
why is this line never hit by tests? should get coverage IMHO

Reply:
The line is tested somewhat here: https://github.com/mne-tools/mne-bids/blob/213c3789390dfacc285418d8f54c6575ca104e7f/mne_bids/tests/test_write.py#L1858-L1866

The reason it is never hit is because other files (like events.tsv) get written earlier than the neural data file in `write_raw_bids`, and these raise an identical error:

https://github.com/mne-tools/mne-bids/blob/213c3789390dfacc285418d8f54c6575ca104e7f/mne_bids/utils.py#L206-L207

Reply:
So … then we can just remove this `else` branch, no?

Reply:
I'd leave it here. People may manually remove all other files and **then** this line would reasonably trigger. it's just not covered in our tests, which I find okay (and which I did not introduce here) :)

Reply:
alright, let's leave it here then!